packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/CanonicalTypeProvider.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Generator.CSharp.Input;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.SourceInput;
+
+namespace Microsoft.Generator.CSharp.Providers
+{
+    internal class CanonicalTypeProvider : TypeProvider
+    {
+        private readonly InputModelType? _inputModel;
+        private readonly TypeProvider _generatedTypeProvider;
+
+        public CanonicalTypeProvider(TypeProvider generatedTypeProvider, InputType? inputType)
```

Comment:
add a todo to fill in the other pieces like methods constructors etc eventually?  Or do we think we will never need those?